lib/physiqual/data_services/google_service.rb

```diff
@@ -53,30 +53,41 @@ def sleep(from, to)
 
       private
 
-      def activity_data(from, to, url, value_type, &block)
+      def activity_data(from, to, url, value_type, &_block)
         res = point_results(from, to, url)
-
-        results = loop_through_results(res) do |value, start, endd, results_hash|
-          actual_timestep = Time.at((start + endd) / 2)
+        loop_through_results(res) do |value, start, endd, results_array|
+          current_value = value[value_type].to_i
+          current_value = [(block_given? ? yield(current_value) : current_value)]
+          measurement_moment = calculate_measurement_moment(start, endd)
 
           # If the current timestep is higher than the final timestep, don't include it
-          next if actual_timestep > to
-          results_hash[actual_timestep] += value[value_type].to_i
+          next if measurement_moment_out_of_range?(measurement_moment, to)
+          results_array << DataEntry.new(start_date: Time.at(start).in_time_zone, end_date: Time.at(endd).in_time_zone,
+                                         values: current_value,
+                                         measurement_moment: measurement_moment)
         end
-
-        hash_to_array(results, &block)
       end
 
-      def specific_activity_data(from, to, url, activity_type, &block)
+      def specific_activity_data(from, to, url, activity_type)
         res = point_results(from, to, url)
-
-        results = loop_through_results(res) do |value, start, endd, results_hash|
-          next unless value['intVal'] == activity_type
-          actual_date = Time.at(endd).in_time_zone.beginning_of_day
-          results_hash[actual_date] += (endd - start) / 60
+        loop_through_results(res) do |value, start, endd, results_array|
+          # If the current activity is not the specified activity, skip it
+          next if value['intVal'] != activity_type
+          measurement_moment = calculate_measurement_moment(start, endd)
+
+          next if measurement_moment_out_of_range?(measurement_moment, to)
+          results_array << DataEntry.new(start_date: Time.at(start).in_time_zone, end_date: Time.at(endd).in_time_zone,
+                                         values: [(endd - start) / 60],
+                                         measurement_moment: measurement_moment)
```

Comment:
Dit stond ook al in de oude code, maar waarom delen we hier `(endd-start)` door `60`?